### PR TITLE
Enable DINO to OTX - Step 2.5. Finalize productization of DINO

### PR DESCRIPTION
**Pull request by jaegukhyun**
### Summary
This PR is for finalizing productization of DINO. 
Some missing parts will be handled next PR(probably next sprint).

**Contents**
- Add DINO and Deformable DETR smart weight loading
- DINO and Deformable DETR configs are merged with incremental recipe  
- Update model size and flops for DINO and Deformable DETR

**Next PR will contain**
-  E2E Tests for Deformable DETR, DINO
-  Add documentation 
- Other OTX CLI support(demo, explain, etc)

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [x] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: otx/algorithms/detection/adapters/mmdet/models/detectors/custom_dino_detector.py [hunk 1]
```diff
@@ -23,17 +23,97 @@ class CustomDINO(CustomDeformableDETR):
...
-    def load_state_dict_pre_hook(ckpt_dict, *args, **kwargs):
+        self.cls_layers.append("dn_query_generator.label_embedding.weight")
+
+    def load_state_dict_pre_hook(self, model_classes, ckpt_classes, ckpt_dict, *args, **kwargs):
         """Modify mmdet3.x version's weights before weight loading."""
 
         if list(ckpt_dict.keys())[0] == "level_embed":
             logger.info("----------------- CustomDINO.load_state_dict_pre_hook() called")
             # This ckpt_dict comes from mmdet3.x
+            ckpt_classes = [
```
Comment (harimkang) [inline otx/algorithms/detection/adapters/mmdet/models/detectors/custom_dino_detector.py:35]: This is a list that will go away with the upgrade to mmdet 3.x, right?

Comment (jaegukhyun) [reply otx/algorithms/detection/adapters/mmdet/models/detectors/custom_dino_detector.py:35]: Yes.

**Review by harimkang — APPROVED** [review]

File: otx/algorithms/detection/adapters/mmdet/models/detectors/custom_dino_detector.py (hunk 1 above)
Comment (eunwoosh) [inline otx/algorithms/detection/adapters/mmdet/models/detectors/custom_dino_detector.py:35]: It's just a question. Loading weight isn't considered in current step?

Comment (jaegukhyun) [reply otx/algorithms/detection/adapters/mmdet/models/detectors/custom_dino_detector.py:35]: No it's loading weights from mmdet3.x and meta information such as classes are located differently with mmdet2.x. Therefore this codes just specify COCO 80 classes

**Review by eunwoosh — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Pull request merged**